### PR TITLE
Thread based GameTimer with diff sleep

### PR DESCRIPTION
**Pull request by ScottHolden**
Thought it would be fun to play around with threads, works quite nicely, but does steal 2 additional threads.

One nice touch is the Thread.Sleep period is kind-of dynamic, it tires to keep the invocation rate as close as possible to Interval. Eg, with an interval of 16ms, and an Invoke of 4ms, it will only sleep for 12ms.

Let me know your thoughts :)

### CODE REVIEW

**Comment by ScottHolden** [comment]
Hold off on this, I'm adding a few tests to verify behavior.

**Comment by ScottHolden** [comment]
Ok, ready to go! 60+ fps with 140 trains and a 14ms Draw time! So butter smooth!
image

Also, look at the minimap, my screen seems to take up nearly the entire world :P

**Comment by ScottHolden** [comment]
This test failure makes no sense!!! It's not even one of the most intensive theories!!! Grrr. Apologies for the discord spam.

File: src/Trains.NET.Engine/Utilities/GameTimer.cs [hunk 1]
```diff
@@ -4,7 +4,8 @@
 
 namespace Trains.NET.Engine
 {
-    public class GameTimer : ITimer
+    // Disable GameTimer :)
```
Comment (davidwengier) [inline src/Trains.NET.Engine/Utilities/GameTimer.cs:7]: Just delete the file, we have git.

**Pull request merged**